File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeWithSensoryEvents.scala
package examples.scala.recipes

import com.ing.baker.recipe.scaladsl.{Event, Recipe}
import examples.scala.events.{FraudCheckCompleted, OrderPlaced, PaymentReceived}

object RecipeWithSensoryEvents {
  val recipe: Recipe = Recipe(name = "example")
    .withSensoryEvents(
      Event[OrderPlaced],
      PaymentReceived.event,
      FraudCheckCompleted.event
    )
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/RecipeWithSensoryEvents.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import examples.kotlin.events.FraudCheckCompleted
import examples.kotlin.events.OrderPlaced
import examples.kotlin.events.PaymentReceived

@ExperimentalDsl
object RecipeWithSensoryEvents {
    val recipe = recipe(name = "example") {
        sensoryEvents {
            eventWithoutFiringLimit<OrderPlaced>()
            event<PaymentReceived>()
            event<FraudCheckCompleted>(maxFiringLimit = 5)
        }
    }
}